### ISSUE TITLE
Issue with push notifications

### ISSUE THREAD

**Issue by ranhsd** [BranchMetrics/ios-branch-deep-linking#703]
Hi, I've managed to integrate Branch in my iOS project for deep linking and referral program and it works great! (I move to Branch from Firebase dynamic links). 

Because i really like how Branch handle links i decided to adopt it also for handle push notifications.
For testing it, i sent push payload from my server to my device. The push payload contains the branch links and it looks like the following: 

```
{
  "alert" : "user test",
  "branch": "https://*****.test-app.link/User/1234"
}
```

***** is my domain.

As you can see. My payload contain a simple alert and the Branch link. This link should route to the user view controller and display user with id of **1234**. 

After sending the push payload i see the **didReceiveRemoteNotification** is being called. This method contains the following code: 

```
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
    
    if ([userInfo objectForKey:@"branch"]) {
        [[Branch getInstance] handlePushNotification:userInfo];
    } else {
        [[NotificationHandler shared] handleRemoteNotification:application.applicationState launchOptions:self.launchOptions userInfo:userInfo fetchCompletionHandler:^(UIBackgroundFetchResult result) {
            completionHandler(result);
        }];
    }
}
```
I see that **handlePushNotification** is being called and then **initSessionWithLaunchOptions** callback is being called and there i have the issue. 

I am using the following callback in my code: 

![image](https://user-images.githubusercontent.com/3685671/30017378-9bd0c4a2-9161-11e7-91ee-4503b27ae99c.png)

 
and when i debug it i see that both **universalObject** and **linkProperties**  are nil

![image](https://user-images.githubusercontent.com/3685671/30017410-be750c5c-9161-11e7-9127-bf16f448addc.png)

I tried to implement it according to the following user guide: 

https://dev.branch.io/getting-started/deep-link-routing/guide/ios/#branch-links-in-push-notifications

but from some reason its not working. 

When i print **[[Branch getInstance] getLatestReferringParams]** i get the following in my debugger: 

```
{
    "+clicked_branch_link" = 0;
    "+is_first_session" = 0;
    "+non_branch_link" = "https://*****.test-app.link/User/1234";
}
```
***** is my domain 

There i see the link but from some reason its say **non_branch_link** 
in the docs it says that you need to use Branch subdomain and route to the relevant source via **/** so what i did is i built the Branch link and i want to navigate to specific user via the following URI: /User/1234

but from some reason it doesn't work.

Do you know what i am doing wrong? 

Thanks in advanced!

**Comment by E-B-Smith** [comment]
I'm glad you like Branch links!

It looks like you're doing everything right.  Let me look into this. Thanks for opening this issue.

**Comment by E-B-Smith** [comment]
I've submitted a PR that fixes this: PR #704 . The fix will be available soon on the master branch, after we QA the release.

**Comment by ranhsd** [comment]
Thanks for this! Will check it on my project and let you know soon.

**Comment by E-B-Smith** [comment]
I fixed this in version 0.18.8.

https://github.com/BranchMetrics/ios-branch-deep-linking/releases/tag/0.18.8
